File: Lab 05/tcp-udp-practice/TCPServer.py
from socket import *
serverPort = 12000
serverSocket = socket(AF_INET,SOCK_STREAM)
serverSocket.bind(('0.0.0.0',serverPort))
serverSocket.listen(1)
print('The server is ready to receive')
while 1:
     connectionSocket, addr = serverSocket.accept()

     sentence = connectionSocket.recv(1024)
     print("TCP Receive", sentence, "FROM", addr)

     capitalizedSentence = sentence.upper()
     connectionSocket.send(capitalizedSentence)
     connectionSocket.close()

File: Lab 05/tcp-udp-practice/golang/TCPServer.go
package main

import (
	"fmt"
	"net"
	"strings"
)

func main() {
	serverPort := 12000
	serverAddress := &net.TCPAddr{
		IP:   net.ParseIP("localhost"),
		Port: serverPort,
	}
	serverSocket, err := net.ListenTCP("tcp", serverAddress)
	if err != nil {
		panic(err)
	}
	defer serverSocket.Close()

	fmt.Println("The server is ready to receive")

	for {
		buffer := make([]byte, 2048)
		connection, err := serverSocket.AcceptTCP()
		if err != nil {
			fmt.Println("Error accepting TCP connection: ", err)
			continue
		}

		connection.Read(buffer)
		message := string(buffer)
		fmt.Printf("TCP receive %s FROM %v\n", message, connection.RemoteAddr())

		messageUpper := strings.ToUpper(message)

		if _, err := connection.Write([]byte(messageUpper)); err != nil {
			fmt.Println("Error sending TCP message: ", err)
			continue
		}
	}
}
